edx_notifications/server/web/static/edx_notifications/js/views/notification_pane_view.js

```diff
@@ -0,0 +1,54 @@
+var NotificationPaneView = Backbone.View.extend({
+    initialize: function(){
+        /* re-render if the model changes */
+        this.listenTo(this.collection, 'change', this.collectionChanged);
+
+        this.hydrate();
+
+        this.render();
+    },
+
+    hydrate: function() {
+        /* This function will load the bound collection */
+
+        /* add and remove a class when we do the initial loading */
+        /* we might - at some point - add a visual element to the */
+        /* loading, like a spinner */
+        var self = this;
+        self.$el.addClass('loading');
+        this.collection.fetch({
+            success: function(){
+                self.$el.removeClass('loading')
+            }
+        });
+    },
+
+    /* cached notification_icon.html template */
+    fetched_template: null,
+
+    collectionChanged: function() {
+        /* redraw for now */
+        this.render();
+    },
+
+    render: function() {
+        if (!this.fetched_template) {
+            var self = this;
+
+            /* load Underscore template asynchronously on first load */
+            $.get("/static/edx_notifications/templates/underscore/notification_pane.html")
+                .done(function(template_data) {
+
+                    /* convert to Underscore template object */
+                    self.fetched_template = _.template(template_data);
+
+                    /* now render template with our model */
+                    self.$el.html(
```

Comment:
Shouldn't this render be outside of the caching logic? It looks like it will only render the first time...

If you do decide to roll your own template loading code, I recommend moving it into a helper method that can be shared across your various files. As it stands, the rendering logic is hard to understand as it is obscured by all the template handling.